### PR TITLE
Documentation improvements in the Configuration section 

### PR DESCRIPTION
**Pull request by asv365**
# Closes # 
This pull request aims to improve the documentation of the Configuration section. It addresses points 3, 4 and 5 of #609. Thanks @ekatef for the feedback.

## Changes proposed in this Pull Request

1. The picture in the lines section includes too many things, the “lines” only appear at the very end.

2. Create subsections in the electricity section, as in the renewables section.

3. The section load from the documentation doesn’t seem to refer to anything on the config file. Shall we remove this?

4. Add a short description in each subsection, as it is done in `run`.

5. General consistency check accross the configuaration section.

## Checklist

- [ ] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by asv365** [comment]
Point 1 is caused by the appearance of the text "lines:" before the lines section header. The lines section header is where the image should start to be displayed. 

For context, the images are created using the instruction:

.. literalinclude:: ../config.default.yaml
   :language: yaml
   :start-at: lines:
   :end-before: links:

Since the text ''lines:'' appears before in the config.default.yaml file (namely in the instruction split_overpassing_**lines:** true), the image displayed is from that first appearance of "lines:" until the first appearance of the word "links:".

Haven't found a good fix yet, but to illustrate the problem the first commit changes "lines:" for "PV:", which is an improvement.

**Comment by ekatef** [comment]
@asv365 A very good point regarding catching text from a wrong fragment. Thanks a lot for investigating it!

Probably, Sphinx docs on [literalinclude](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude) could be helpful to resolve this, eventually combined with other Shpinx magic. The most straightforward way seems to be use tag comments. But I suspect there might be a more elegant solution, as well.

**Comment by asv365** [comment]
Thanks @ekatef, the suggestion on comments can be a good workaround if we can't find how to do it properly with Sphinx. If Sphinx proves tricky, we could add a header to every section of the config file, to be consistent.

**Comment by asv365** [comment]
On point 3., the section "load" has been renamed "operational_reserve", and has been placed as a subsection of electricity. There is also the section "load_options", which doesn't exist in the documentation. This will be addressed as point 5, as part of the general consistency check.

**Comment by asv365** [comment]
@ekatef  found a solution to the point 1 we were discussing earlier without the need to use comments in the config file. 

Literalinclude allows the command `start-after` as well as `start-at`. `start-after` ignores the line you are pointing to, and starts displaying in the next one. Since the previous line is `PV:`, which is unique, it works for this case.

**Comment by ekatef** [comment]
> @ekatef found a solution to the point 1 we were discussing earlier without the need to use comments in the config file.
> 
> Literalinclude allows the command `start-after` as well as `start-at`. `start-after` ignores the line you are pointing to, and starts displaying in the next one. Since the previous line is `PV:`, which is unique, it works for this case.

@asv365 a very nice trick which looks absolutely in spirit of Sphynx! :) 

Probably, there is a little risk to catch an additional section in case it would be inserted between `estimate_renewable_capacities` and `lines`...  Could it make sense to add a comment to `configuration.rst` as a reminder on why do we refer exactly `PV`?

Although, the work you are doing is still a great improvement for clarity of the docs 🥇

**Comment by asv365** [comment]
Good point @ekatef. I'll add that comment in `configuration.rst`

**Comment by asv365** [comment]
@ekatef The top-level configuration section (which, according to the table in the docs includes "version", "tutorial", "logging", "countries", and "enable") has other sections inbetween. For example, "run" and "scenarios" are between "logging" and "countries" in the `config.default.yaml`. 

Do we want to bring all the items of the top-level configuration together at the top of the `config.default.yaml` and `config.tutorial.yaml`? Not doing that forces literalinclude to display using line numbers, which was creating trouble earlier. See, e.g., commit [d71a3a5](https://github.com/pypsa-meets-earth/pypsa-earth/pull/612/commits/d71a3a5cc73d60df5ca8da0214335c3ecc23cfee) above. 

Another option would be to include those intermediate sections as part of the top-level configuration. Let me know what you think :).

**Comment by ekatef** [comment]
@asv365, thanks for rising this point! Our current `top-level configuration` is in fact rendered in not optimal way...

Completely agree that it might be worth to revise underlying assumptions for making the result more robust and the docs more convenient. I feel that it is a very good point for discussion during developers meeting :) Would you mind to introduce this topic tomorrow?

**Comment by asv365** [comment]
Thank you @ekatef. I'll raise it in tomorrow's meeting :)

**Comment by asv365** [comment]
In the pypsa-earth weekly meeting on 02/03/2023, it was decided that:

1. In literalinclude, preference of "start-at" like command instead of hard-coded line numbers.
2. The top-level configuration section should include the sections that are already there. Reshuffling in config.default.yaml allowed, but keep to the bare minimum and report it in the PR. Another option is to include several "literalincludes".
3. Make the Value column as short as possible, and very symbolic.

File: doc/configuration.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Top-level configuration
 
 .. literalinclude:: ../config.default.yaml
    :language: yaml
-   :lines: 5-12,20,31-38
+   :lines: 5-12,24,31-38
```
Comment (pz-max) [inline doc/configuration.rst:28]: I think we will change that by the `start-at` and `end-at` stuff?

**Review by pz-max — COMMENTED** [review]
pre-commit needs fixes as it could not automatically get fixed.
How far are we from merging this  PR @asv365 ?

File: doc/configuration.rst (hunk 1 above)
Comment (asv365) [reply doc/configuration.rst:28]: yes @pz-max, that is the plan. Just did this to motivate the `start-at` and `end-at` approach. Will work through this soon I hope, a lot of work has popped up recently.

**Comment by ekatef** [comment]
Hey @asv365! Very nice work so far :)
Could you please give an update on the PR status? Don't hesitate to let us know if you need any additional inputs. Would be happy to support you if needed

**Comment by asv365** [comment]
Hi @ekatef, everything is good, thanks for asking and for offering help. I don't think there are bottlenecks so far.

The reason for being so slow are job applications. Will come back to this in the next week or the one after hopefully. The dust is settling now. Apologies for the delay.

**Comment by ekatef** [comment]
@asv365, no worries! Good luck with your applications and come back once you'd have more time ;)

**Comment by asolavi** [comment]
This pull request is being addressed in PR #694 because a fresh restart was useful.

**Comment by davide-f** [comment]
Hello :)
I'm closing this PR as it is continued into the other PR.
Thanks for the contribution!

**Comment by asolavi** [comment]
> Hello :) I'm closing this PR as it is continued into the other PR. Thanks for the contribution!

Thank you, yes!

**Pull request closed without merging**